[DOCU-2418] Kong.conf config (#4186)
Kong/docs.konghq.com: src/gateway/kong-production/kong-conf.md

---
title: Kong Configuration File
content-type: how-to
---

> Kong comes with a default configuration file `kong.conf`. If you installed Kong using an official package, this file can be found at:
> `/etc/kong/kong.conf.default`. The Kong configuration file is a YAML file that can be used to configure individual properties of your Kong instance. This guide will explain how to configure Kong using the `kong.conf` file.

Reviewer: ```suggestion
{{site.base_gateway}} comes with a default configuration file `kong.conf`. If you installed {{site.base_gateway}} using an official package, this file can be found at:
`/etc/kong/kong.conf.default`. The {{site.base_gateway}} configuration file is a YAML file that can be used to configure individual properties of your Kong instance. This guide will explain how to configure {{site.base_gateway}} using the `kong.conf` file.
```

> Kong comes with a default configuration file `kong.conf`. If you installed Kong using an official package, this file can be found at:
> `/etc/kong/kong.conf.default`. The Kong configuration file is a YAML file that can be used to configure individual properties of your Kong instance. This guide will explain how to configure Kong using the `kong.conf` file.

Reviewer: ```suggestion
{{site.base_gateway}} comes with a default configuration file `kong.conf`. If you installed {{site.base_gateway}} using an official package, this file can be found at:
`/etc/kong/kong.conf.default`. The {{site.base_gateway}} configuration file is a YAML file that can be used to configure individual properties of your Kong instance. This guide will explain how to configure {{site.base_gateway}} using the `kong.conf` file.
```

> ## Configure Kong
> 
> To configure Kong, make a copy of the default configuration file: 

Reviewer: ```suggestion
## Configure {{site.base_gateway}}

To configure {{site.base_gateway}}, make a copy of the default configuration file: 
```

                                    # connection.
```

To configure a property, uncomment it and modify the value:

```bash
upstream_keepalive_pool_size = 40
```

Boolean values can be specified as `on`/`off` or `true`/`false`:

```bash
#dns_no_sync = off               # If enabled, then upon a cache-miss every
                                 # request will trigger its own dns query.
                                 # When disabled multiple requests for the
                                 # same name/type will be synchronised to a
                                 # single query.
```

{:.note}
> Kong will use the default settings for any value in `kong.conf` that is commented out.

## Verify configuration
To verify that your configuration is usable, use the `check` command. The `check` command will evaluate the [environment variables](/gateway/latest/kong-production/environment-variables) you have
currently set, and will output an error if your settings are invalid. 

```bash
kong check /etc/kong/kong.conf
```
If your configuration is valid the shell will output:

```bash
configuration at /etc/kong/kong.conf is valid
```

## Set custom path

> By default, Kong looks for `kong.conf` in two

Reviewer: ```suggestion
By default, {{site.base_gateway}} looks for `kong.conf` in two
```
